patches/server/0727-make-configured-features-seed-configurable.patch

```diff
@@ -0,0 +1,81 @@
+         // TODO: Figure out why getFloat() always returns the default value.
+         return (float) getDouble(path, (double) def);
+@@ -820,5 +826,18 @@ public class PaperWorldConfig {
+     private void fixInvulnerableEndCrystalExploit() {
+         fixInvulnerableEndCrystalExploit = getBoolean("unsupported-settings.fix-invulnerable-end-crystal-exploit", fixInvulnerableEndCrystalExploit);
+     }
++
++    public Map<String, Long> configuredFeatureSeedReplacements = new HashMap<>();
++    private void setupConfiguredFeatureSeedReplacer() {
++        String path = "configured-feature.seed-replacements";
++        getLong(path + ".minecraft:disk_clay", 748896458);
++        ConfigurationSection values = config.getConfigurationSection("world-settings." + worldName + "." + path);
++        if (values == null) {
++            values = config.getConfigurationSection("world-settings.default." + path);
++        }
++        for (String key : values.getKeys(false)) {
++            configuredFeatureSeedReplacements.put(key, values.getLong(key));
++        }
++    }
+ }
+ 
+diff --git a/src/main/java/net/minecraft/world/level/biome/Biome.java b/src/main/java/net/minecraft/world/level/biome/Biome.java
+index a7a7e6cd87270e64a92448f03f8b0b0c7e375ec7..11867efa3870113549a15b0fc627aa4703bf567e 100644
+--- a/src/main/java/net/minecraft/world/level/biome/Biome.java
++++ b/src/main/java/net/minecraft/world/level/biome/Biome.java
+@@ -225,8 +225,10 @@ public final class Biome {
+         Registry<StructureFeature<?>> registry2 = region.registryAccess().registryOrThrow(Registry.STRUCTURE_FEATURE_REGISTRY);
+         int i = GenerationStep.Decoration.values().length;
+ 
++        long populationCopy = populationSeed; // Paper - keep a copy here so that any replacements don't override into features that don't need to be changed
+         for(int j = 0; j < i; ++j) {
+             int k = 0;
++            populationSeed = populationCopy; // Paper also don't want configuredFeatures to pollute StructureFeatures
+             if (structureAccessor.shouldGenerateFeatures()) {
+                 for(StructureFeature<?> structureFeature : this.structuresByStep.getOrDefault(j, Collections.emptyList())) {
+                     random.setFeatureSeed(populationSeed, k, j);
+@@ -263,6 +265,16 @@ public final class Biome {
+                     Supplier<String> supplier3 = () -> {
+                         return registry.getResourceKey(configuredFeature).map(Object::toString).orElseGet(configuredFeature::toString);
+                     };
++                    // Paper start - set the populationSeed back to the one defined by the region and if applicable, replace
++                    populationSeed = populationCopy;
++                    ResourceLocation location = registry.getKey(configuredFeature);
++                    Long replacementSeed = region.getMinecraftWorld().paperConfig.configuredFeatureSeedReplacements.getOrDefault(location.toString(), null);
```

Comment:
Might be better to use `ResourceLocation`s in the config Map so we don't need to constantly toString.